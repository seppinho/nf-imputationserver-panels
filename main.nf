#!/usr/bin/env nextflow
/*
========================================================================================
    genepi/nf-imputationserver-refpanels
========================================================================================
    Github : https://github.com/genepi/nf-imputationserver-panels
    Author: Sebastian Sch√∂nherr / Lukas Forer
    ---------------------------
*/

nextflow.enable.dsl = 2


include { NF_IMPUTATIONSERVER_PANELS } from './workflows/nf_imputationserver_panels'

workflow {
    NF_IMPUTATIONSERVER_PANELS ()
}

