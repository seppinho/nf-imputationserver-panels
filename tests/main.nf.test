nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should create three shuffeled msav files") {

        when {
            params {
                project  = "test-job"
                mode = "recom"
                files = "$projectDir/tests/input/three/*.vcf.gz"
            }   
        }

        then {
            assert snapshot(workflow.trace).match()
            assert workflow.success
        }

    }

    test("Should create three msav files from m3vcf") {

        when {
            params {
                project  = "test-job2"
                mode = "convert"
                files = "$projectDir/tests/input/m3vcf/*.m3vcf.gz"
            }   
        }

        then {
            assert snapshot(workflow.trace).match()
            assert workflow.success
        }

    }


}