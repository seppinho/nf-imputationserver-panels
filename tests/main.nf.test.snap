{
    "Should create three msav files": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 3,
                "tasksSucceeded": 3
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2023-11-10T17:02:16.158553"
    },
    "Should create three shuffeled msav files": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 3,
                "tasksSucceeded": 3
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-21T17:04:11.480188"
    },
    "Should create three msav files from m3vcf": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 2,
                "tasksSucceeded": 2
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-21T17:10:29.96432"
    }
}